lock-api/src/main/java/com/palantir/lock/v2/BatchedStartTransactionResponse.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.v2;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableBatchedStartTransactionResponse.class)
+@JsonDeserialize(as = ImmutableBatchedStartTransactionResponse.class)
+public interface BatchedStartTransactionResponse {
+    @Value.Parameter
+    LockImmutableTimestampResponse immutableTimestamp();
+
+    @Value.Parameter
+    TimestampRangeAndPartition startTimestamps();
+
+    @Value.Parameter
+    Lease lease();
+
+    static BatchedStartTransactionResponse of(
+            LockImmutableTimestampResponse immutableTimestamp,
+            TimestampRangeAndPartition startTimestamp,
+            Lease lease) {
+        return ImmutableBatchedStartTransactionResponse.builder()
+                .immutableTimestamp(immutableTimestamp)
+                .startTimestamps(startTimestamp)
+                .lease(lease)
+                .build();
```

Comment:
sometimes it is easier to read when builder is used, but I don't think it is the case with this one. I will change it to use `of()`

Reply:
I've never really found `@Value.Parameter` useful tbh, because we never really use the `of` method of the `Immutable*` class, so we're forced to define the `of` class inside our class. If we then call `Immutable*.of` then it just becomes like calling a big constructor. 

🤷‍♂️

Reply:
Not sure if it's really that we don't use `Immutable*.of` actually. Agree that calling ImmutableX.of() when X.of() exists is strange, though we aren't always consistent with declaring that

```
$ find . | grep "\.java" | grep -v "generated" | xargs cat | grep "Immutable.*\.of(" | grep -vE "Immutable(Sorted)?(Set|Map|List|Multimap)" | grep -o "Immutable[a-zA-Z0-9]*" | sort | uniq -c
  16 ImmutableCallbackInvocation
   3 ImmutableCheckAndSetResult
   1 ImmutableDedicatedRows
   2 ImmutableFastFailoverState
   3 ImmutableIdentifiedLockRequest
   1 ImmutableIdentifiedSystemTime
   4 ImmutableIdentifiedTimeLockRequest
   1 ImmutableLeaderTime
   1 ImmutableLeadershipId
   2 ImmutableLeadershipState
   1 ImmutableLease
   3 ImmutableLockImmutableTimestampResponse
   3 ImmutableLockRequest
   3 ImmutableLockRequestSpecification
   3 ImmutableLockResponse
   2 ImmutableLockToken
   1 ImmutableLockWithClient
   8 ImmutableMaskedValue
   1 ImmutableOrderedLocks
   2 ImmutableOverflowValue
   2 ImmutablePaxosLong
   1 ImmutablePaxosValueResponse
   2 ImmutablePersistentLockId
   1 ImmutableRangeAndValue
   6 ImmutableRangeMap
   2 ImmutableReferenceCountedResidue
   2 ImmutableRefreshLockResponseV2
   2 ImmutableRequestInfo
  11 ImmutableSequenceAndBound
   1 ImmutableSimpleLockRequest
   1 ImmutableSimpleTokenInfo
   2 ImmutableStartAtlasDbTransactionResponse
   1 ImmutableStartAtlasDbTransactionResponseV3
   2 ImmutableStartIdentifiedAtlasDbTransactionRequest
   3 ImmutableStartIdentifiedAtlasDbTransactionResponse
   1 ImmutableStoredWriteReference
   2 ImmutableSweepTableIdentifier
   1 ImmutableTimeLimit
   3 ImmutableTimeLimiterConfiguration
   1 ImmutableTimestamp
   2 ImmutableTimestampAndPartition
   1 ImmutableTimestampClientConfig
   2 ImmutableTimestampPair
   2 ImmutableTimestampPartitioningMap
  11 ImmutableTimestampResponse
   9 ImmutableTodo
   1 ImmutableTsLock
   3 ImmutableUnsuccessful
   2 ImmutableValueAndBound
   3 ImmutableWaitForLocksRequest
   3 ImmutableWaitForLocksResponse
```